epoch.go

```diff
@@ -1890,13 +1906,18 @@ func (e *Epoch) verifyProposalIsPartOfOurChain(block Block) bool {
 	// If it's the former, we need to find the parent of the block and ensure it is correct.
 	if bh.Seq > 0 {
 		// TODO: we should cache this data, we don't need the block, just the hash and sequence.
-		_, found := e.locateBlock(bh.Seq-1, bh.Prev[:])
+		_, found, err := e.locateBlock(bh.Seq-1, bh.Prev[:])
+		if err != nil {
+			e.Logger.Error("Failed locating parent block", zap.Uint64("seq", bh.Seq-1), zap.Error(err))
```

Comment:
I don't think we should log this with an error, this can just be a block that is not connected to the chain, can't it?